src/events/index.ts

```diff
@@ -65,6 +66,85 @@ export class EVENTS extends Base {
     }
   }
 
+  async createUserEvent(
+    events: {
+      event_type: string,
+      event: {
+        event_details: Object,
+        [key: string]: any;
+      }
+    }[],
+  ): Promise<{ message: string }> {
+    await this.refreshSessionIdFromStorage();
+
+    if (!this.sessionID) {
+      throw new Error('Could not create event. No session id. Please initiate a session first (See Helika Docs).');
+    }
+
+    if (!this.userDetails.user_id) {
+      console.error('Cannot create user event without populating user_id using sdk.setUserDetails() function')
+    }
+
+    let created_at = new Date().toISOString();
+    let helika_referral_link: any = null;
+    let utms: any = null;
+    let current_url: string = "";
+    let event_source = 'server';
+    try {
+      if (ExecutionEnvironment.canUseDOM) {
+        helika_referral_link = this.refreshLinkId();
+        utms = this.refreshUtms();
+        current_url = window.location.href;
+        event_source = 'client'
+      }
+    } catch (e) {
+      console.error(e);
+    }
+
+
+    let newEvents = events.map((event: any) => {
+      let givenEvent: any = Object.assign(
+        {},
+        event,
+        {
+          app_details: Object.assign({}, event.app_details, this.appDetails),
+          user_details: Object.assign({}, event.user_details, this.userDetails),
+          helika_data: this.getDeviceDetails()
+        }
+      );
+      givenEvent.event.helika_referral_link = helika_referral_link;
+      givenEvent.event.utms = utms;
+      givenEvent.event.url = current_url;
+      if (event.event.session_id) {
+        givenEvent.event.client_session_id = event.event.session_id
+      }
+      givenEvent.event.session_id = this.sessionID;
+      givenEvent.event.player_id = this.playerId;
+      givenEvent.created_at = created_at;
+      givenEvent.game_id = this.gameId;
+      return givenEvent;
+    });
+
+    var params: {
+      id: string,
+      user_id: string,
+      events: {
+        created_at: string,
+        game_id: string,
+        event_type: string,
+        event: Object
+      }[]
+    } = {
+      id: v4(),
+      user_id: this.userDetails.user_id,
+      events: newEvents
+    }
+
+    this.extendSession();
+
+    return this.postRequest(`/game/game-event`, params);
+  }
+
   async createEvent(
```

Comment:
createEvent() should spit out the exact same taxonomy as the createUserEvent(), the only difference is the user_details is required to get passed along and `user_id` is checked to exist and not be null. 

you can probably turn it into a helper function and call it from createUserEvent()